internal/staging/stdaddr/address_test.go

```diff
@@ -0,0 +1,457 @@
+				t.Errorf("%s: mismatched change script -- got %x, want %x",
+					test.name, gotScript, wantScript)
+				continue
+			}
+
+			// Ensure the vote commitment script version and contents are the
+			// expected values.
+			wantScript, err = hex.DecodeString(test.commitScript)
+			if err != nil {
+				t.Errorf("%s: unexpected hex decode err: %v", test.name, err)
+				continue
+			}
+			gotScriptVer, gotScript = stakeAddr.PayVoteCommitmentScript()
+			if gotScriptVer != test.version {
+				t.Errorf("%s: mismatched vote commit script version -- got %d, "+
+					"want %d", test.name, gotScriptVer, test.version)
+				continue
+			}
+			if !bytes.Equal(gotScript, wantScript) {
+				t.Errorf("%s: mismatched vote commit script -- got %x, want %x",
+					test.name, gotScript, wantScript)
+				continue
+			}
+
+			// Ensure the revoke commitment script version and contents are the
+			// expected values.
+			wantScript, err = hex.DecodeString(test.revokeScript)
+			if err != nil {
+				t.Errorf("%s: unexpected hex decode err: %v", test.name, err)
+				continue
+			}
+			gotScriptVer, gotScript = stakeAddr.PayRevokeCommitmentScript()
+			if gotScriptVer != test.version {
+				t.Errorf("%s: mismatched revoke cmt script version -- got %d, "+
+					"want %d", test.name, gotScriptVer, test.version)
+				continue
+			}
+			if !bytes.Equal(gotScript, wantScript) {
+				t.Errorf("%s: mismatched revoke cmt script -- got %x, want %x",
+					test.name, gotScript, wantScript)
+				continue
+			}
+
+			// Ensure the pay from treasury script version and contents are the
+			// expected values.
+			wantScript, err = hex.DecodeString(test.trsyScript)
+			if err != nil {
+				t.Errorf("%s: unexpected hex decode err: %v", test.name, err)
+				continue
+			}
+			gotScriptVer, gotScript = stakeAddr.PayFromTreasuryScript()
+			if gotScriptVer != test.version {
+				t.Errorf("%s: mismatched treasury change script version -- "+
+					"got %d, want %d", test.name, gotScriptVer, test.version)
+				continue
+			}
+			if !bytes.Equal(gotScript, wantScript) {
+				t.Errorf("%s: mismatched treasury change script -- got %x, "+
+					"want %x", test.name, gotScript, wantScript)
+				continue
+			}
+		}
+
+		// Ensure encoding the address is the same as the original.
+		encoded := decodedAddr.Address()
+		if encoded != test.addr {
+			t.Errorf("%s: decoding and encoding produced different addresses "+
+				"-- got %v, want %v", test.name, encoded, test.addr)
+			continue
+		}
+
+		// Ensure the stringer returns the same address as the original.
+		if ds, ok := decodedAddr.(fmt.Stringer); ok && ds.String() != test.addr {
+			t.Errorf("%s: mismatched decoded stringer -- got %v, want %v",
+				test.name, ds.String(), test.addr)
+			continue
+		}
+
+		// Ensure the Hash160 method for the addresses that support it returns
+		// the expected value.
+		if h160er, ok := decodedAddr.(Hash160er); ok {
+			decodedBytes := base58.Decode(test.addr)
+			wantH160 := decodedBytes[2 : 2+ripemd160.Size]
```

Comment:
Yes, that's correct.  I'm making the assumption that hash160 is going away with new versions since that is the current plan.  If that ends up not being the case, these tests will need to be adjusted accordingly as you note.